src/lfortran/semantics/ast_common_visitor.h

```diff
@@ -4132,10 +4132,12 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                         ASR::Array_t* rhs_array
                             = ASR::down_cast<ASR::Array_t>(ASRUtils::expr_type(tmp_init));
                         if (lhs_array->n_dims != rhs_array->n_dims) {
-                            diag.add(Diagnostic("Incompatible ranks in assignment",
+                            diag.add(Diagnostic(
+                                "Incompatible ranks `"+ std::to_string(lhs_array->n_dims) + "` and `"
+                                                                  + std::to_string(rhs_array->n_dims) + "` in assignment",
```

Comment:
```suggestion
                                "Incompatible ranks "+ std::to_string(lhs_array->n_dims) + " and "
                                                                  + std::to_string(rhs_array->n_dims) + " in assignment",
```

Let's not use backticks for numbers. We use them for highlighting symbol names like variables.